src/DependencyEmitter/FQDNIndexNode.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\DependencyEmitter;
+
+class FQDNIndexNode
+{
+    private bool $FQDN = false;
+
+    /**
+     * @var array<string, FQDNIndexNode>
+     */
+    private array $nodes = [];
+
+    public function isFQDN(): bool
+    {
+        return $this->FQDN;
+    }
+
+    /**
+     * @param string[] $keys
+     */
+    public function getNestedNode(array $keys): ?self
+    {
+        $index = $this;
+        foreach ($keys as $key) {
+            $node = $index->nodes[$key] ?? null;
+            if (null === $node) {
+                return null;
```

Comment:
How about keeping the previous node and skip the null check in the usage below?

Something like this:
```php
    public function getNestedNode(array $keys): ?self
    {
        $index = $previous = $this;
        foreach ($keys as $key) {
            $node = $index->nodes[$key] ?? null;
            if (null === $node) {
                return $previous;
            }

            $index = $previous = $node;
        }

        return $index;
    }
```

and then:
```php
    protected function isFQDN(AstDependency $dependency, FQDNIndexNode $FQDNIndex): bool
    {
        $dependencyFQDN = $dependency->getTokenName()->toString();
        $path = explode('\\', $dependencyFQDN);

        return $FQDNIndex->getNestedNode($path)->isFQDN();
    }
```

Reply:
Hm, I expect to receive exactly what I requested, when I call the method with name `getNestedNode`. In this case method should be called `getClosesdNode` or something like this. What do you think about it?

Reply:
I have different results

Before:
```
 -------------------- -------- 
  Report                       
 -------------------- -------- 
  Violations           0       
  Skipped violations   641     
  Uncovered            3891    
  Allowed              114698  
  Warnings             0       
  Errors               0       
 -------------------- -------- 
```

With your suggestion:
```
 -------------------- -------- 
  Report                       
 -------------------- -------- 
  Violations           0       
  Skipped violations   641     
  Uncovered            2638    
  Allowed              114614  
  Warnings             0       
  Errors               0       
 -------------------- -------- 
```

Vanila 0.19.1
```
 -------------------- -------- 
  Report                       
 -------------------- -------- 
  Violations           0       
  Skipped violations   641     
  Uncovered            3871    
  Allowed              114698  
  Warnings             0       
  Errors               0       
 -------------------- -------- 
```

Reply:
If my suggestion throws it off this much, feel free to ignore it. I was just bothered by the nullable return, but it's not _that_ important. I'm still confused about the difference between this PR and 0.19.1. Do you have an idea why this is?